content/blog/simplify-embedded-projects-with-esp-bsp/index.md

```diff
@@ -0,0 +1,238 @@
+   ```
+
+### Exploring the Example
+
+Once the application is running, you’ll see the following features in action:
+
+- **Display**: Shows images, text files, and more.
+- **Touch**: Interacts with the display.
+- **Audio**: Plays sound files.
+
+Let's look at the source code of the example. The Board Support Package provides an API that allows you to initialize a board. The application code is cleaner and does not require board-specific details, such as pins hard-coded in the application code.
+
+```c
+#include "esp_log.h"
+#include "bsp/esp-bsp.h"
+#include "app_disp_fs.h"
+
+static const char *TAG = "example";
+
+void app_main(void)
+{
+    /* Initialize and mount SPIFFS */
+    bsp_spiffs_mount();
+
+    /* Initialize I2C (for touch and audio) */
+    bsp_i2c_init();
+
+    /* Initialize display and LVGL */
+    bsp_display_start();
+
+    /* Set default display brightness */
+    bsp_display_brightness_set(APP_DISP_DEFAULT_BRIGHTNESS);
+
+    /* Add and show LVGL objects on display */
+    app_disp_lvgl_show();
+
+    /* Initialize SPI flash file system and show list of files on display */
+    app_disp_fs_init();
+
+    /* Initialize audio */
+    app_audio_init();
+
+    ESP_LOGI(TAG, "Example initialization done.");
+}
+```
+
+### Simulating the ESP32-S3-BOX-3 with Wokwi
+
+You can also simulate the application with ESP-BSP for ESP32-S3-BOX-3 using the Wokwi simulator from your development environment using a plugin for [VS Code](https://docs.wokwi.com/vscode/getting-started), [Espressif IDE](https://github.com/espressif/idf-eclipse-plugin?tab=readme-ov-file#wokwi-simulator), or [JetBrains IDEs](https://plugins.jetbrains.com/plugin/23826-wokwi-simulator). This allows you to test your applications without needing the physical hardware.
+
+To set up the simulation, you need to add two files to your project: [wokwi.toml](wokwi.toml) and [diagram.json](diagram.json) in the root of the project.
+
+Content of `wokwi.toml`
+```toml
+[wokwi]
+version = 1
+elf = "build/uf2.bin"
+firmware = "build/uf2.bin"
+```
+
+Content of `diagram.json`
+```json
+{
+  "version": 1,
+  "author": "Wokwi.com",
+  "editor": "wokwi",
+  "parts": [
+    {
+      "type": "board-esp32-s3-box-3",
+      "id": "esp32",
+      "top": -24.91,
+      "left": -388.54,
+      "attrs": { "psramSize": "16", "flashSize": "16" }
+    }
+  ],
+  "connections": [ [ "$serialMonitor:RX", "esp32:G14", "", [] ], [ "$serialMonitor:TX", "esp32:G11", "", [] ] ],
+  "dependencies": {}
+}
+```
+
+Use the following command to build the UF2 image for your project. The Wokwi simulator will pick up this binary and run it.
+
+```bash
+idf.py uf2
+```
+
+Now you can start the simulation in your favorite IDE.
+
+## Conclusion
+
+With ESP-BSP, you can quickly develop and port your applications across various ESP32 boards, saving time and effort. Whether you’re building a new project or upgrading an existing one, ESP-BSP simplifies your development process.
```

Comment:
@georgik Please consider replacing `ESP32 boards` as per the comment above.

All other changes, LGTM!

Reply:
Replaced by Espressif SoC.